synapse/lib/cell.py

```diff
@@ -2325,6 +2330,63 @@ def getArgParser(cls, conf=None):
 
         return pars
 
+    @classmethod
+    async def provAhaSvc(cls, opts, conf):
+
+        donepath = s_common.genpath(opts.dirn, 'prov.done')
+        if os.path.isfile(donepath):
+            return
+
+        provurl = conf.get('aha:provision')
+        if provurl is None:
+            return
+
+        certdir = s_certdir.CertDir(path=s_common.gendir(opts.dirn, 'certs'))
+
+        confedit = {}
+        async with await s_telepath.openurl(provurl) as prov:
+
+            certdir.saveCaCertByts(await prov.getCaCert())
+
+            provinfo = await prov.getProvInfo()
+
+            name = provinfo.get('name')
+            urls = provinfo.get('urls')
+            netw = provinfo.get('network')
+            admin = provinfo.get('admin')
+
+            if isinstance(urls, str):
+                urls = (urls,)
+
+            registry = []
+            for url in urls:
+                urldict = s_urlhelp.chopurl(url)
+                urldict['user'] = name
+                registry.append(s_telepath.zipurl(urldict))
+
+            hostname = f'{name}.{netw}'
+            username = f'{name}@{netw}'
+
+            confedit['aha:name'] = hostname
+            confedit['aha:network'] = netw
+            confedit['aha:admin'] = provinfo.get('admin')
+            confedit['aha:registry'] = registry
+            confedit['dmon:listen'] = f'ssl://0.0.0.0:0?hostname={hostname}&ca={netw}'
```

Comment:
Do we want to support alternative hostnames here?
It's not unreasonable [but not common] for a service to specify a separate hostname for itself that isn't part of the aha network/

Reply:
The only thing that specifying a hostname for the bind address does is force it to bind only the IP address that the hostname resolves to at the moment it starts up, so we dont need to worry about that.  If they really want to they can edit that in ex-post-facto.